Prevent vue from reusing project page component

lynxplay opened the pull request:
Previously vue would reuse the project page component when switching
between project pages as the component was not keyed.
This commit fixes this by partially modifying the project page setup.

In the current state both the project page content as well as the list
of opened project pages, used to properly configure the tree views
opened elements, are handled in the same state.
This layout however introduces a rather unfortunate flaw into the logic
which makes this fix a bit larger than a simple :key value.

The openedProjectPages `open` contains state that is expected to outlive
a simple change from one project page to another to properly show the
project page tree. This directly conflicts with the assumptions made
about the rest of the project state, which should be discarded the
displayed project page is changed as content and potential edit values
are no longer applicable.

To implement this, this commit splits the opened project pages array
into its own composable as well as the project page markdown into its
own component.

The opened project pages are then tracked by the owning route, e.g. the
prjects index route or the general page route, while the current page
markdown lives in a new component which owns its own state and is keyed
with the pages path.
This way, a change in route properly re-creates the project page
markdown component while keeping the same opened page state in the
parent component.

# Comments

This could also be solved with less diff by actively updating the page content on route change, however I personally believe it does not make much sense to keep the page markdown state in the parent component of either the [...all].vue or the index.vue.

kennytv commented:
https://streamable.com/te3bro

kennytv commented:
Although I can confirm the bad hydration error on opening projects in a new tab as well as the doc subpage switch issues are fixed with this - just the video above is a regression where an empty description and sponsors card appears when you switch from the main docs page to another tab.

lynxplay commented:
I fixed it up into the commit, (for future reference, should I fix up these or single commits to make it easier for testing tho harder for merging as those single line change commits would have to be squashed).

I removed the key from the project index main doc field, preventing it from being re-evaluated when the nav switches.

kennytv commented:
Adding new commits is better for review, since here it's always squashed

kennytv reviewed (approved):
Very hot, thank you

kennytv commented:
Okay well the hydration issue remains, but other project page opening issues were fixed*

lynxplay commented:
I'll look at the hydration issue after I am done with #713 :+1:

lynxplay commented:
Well, I did look further into the hydration issue. The issue certainly replicable on my end somewhat easily on any page of hangar for that matter, not just a specific profile. Here are the steps:

Create an href to a hangar page (in e.g. a gist). Such as http://localhost:3333.
Click it.
Hydration error :+1: 

On the root index, all hydration errors happen in the Header component, on profiles/projects this gets even further out of hand with a bunch of components failing hydration.
Might be worth opening a new issue for this ?
